### ISSUE TITLE
Refactoring Widgets names

### ISSUE THREAD

**Issue by aberba** [dayllenger/beamui#14]
So upon browsing through the api docs and could find a text input. Its appears its called EditLine by looking through code samples. I remember dlangui had some these weird names but you'd probably agree we now commonly refer to them as TextInput, TextEntry, TextBox, etc in mainstream UI. 

Have you thought about changing some of the names to better fit familiar terms?

**Comment by dayllenger** [comment]
Yes, text fields certainly should be renamed. If you encounter other unintuitive names, write them here, I'll change some of them.

**Comment by aberba** [comment]
Oh ok. Will do that and get back to you

**Comment by aberba** [comment]
Here are some names you might consider

```d
platform.enterMessageLoop() -> platform.runMessageLoop();

TabWidget -> Tab
EditLine -> TextInput
EditBox -> TextArea
CanvasWidget -> Canvas
ImageWidget -> Image // cus its already under the Widgets module. 
```

Those are some few suggestions now. Will add more when I encounter any. 

As is stands, the api is quite clean, reading from the examples dir.

**Comment by dayllenger** [comment]
Ok, so I've done:
```d
platform.enterMessageLoop() -> platform.runEventLoop();

TabWidget -> Tabs
EditLine -> TextField   // as in Flutter, MaterialUI, SwiftUI, UIKit, JavaFX...
EditBox -> TextArea
CanvasWidget -> Canvas
```
`Image` is a reserved name for lower-level API for now. I haven't decided yet where I will use it.

**Comment by aberba** [comment]
Great. Thanks.

 Also wanted to know if the UI is reactive like flutter or react. It didn't seem so just by looking at the code. If not, do you think it'll be an interesting thing to do or you'd recommend a third-party lib like rx-d

**Comment by dayllenger** [comment]
Yes, it is. Old widgets are now called "elements", and new widgets work upon them, somewhat like react components or flutter widgets. I still have to port about half of old widgets onto new system.

**Comment by aberba** [comment]
Nice.

**Comment by aberba** [comment]
Why is beam ui not in the dub repository already?

**Comment by dayllenger** [comment]
Because I made major API reconstructions in the past and I still need to do some more. Coming soon...

**Comment by aberba** [comment]
What do you think remains to be done to have a beta version of some kind out?

**Comment by dayllenger** [comment]
Hi. I was super busy last two months, now a little less busy, slowly working for release.

I need to do these things:
- popups and dialogs - they don't really work right now
- virtualized lists/grids - like in [react-window](https://github.com/bvaughn/react-window), but maybe even simpler
- CSS font selection algorithm and `@font-face` - critical for CJK, emoji, icon fonts, any fancy characters
- some very simple UI inspector
- media queries
- and some means to render UI to a texture

**Comment by aberba** [comment]
Glad you're back.
